packages/driver/src/cy/commands/actions/type.ts

```diff
@@ -360,9 +360,10 @@ export default function (Commands, Cypress, cy, state, config) {
               // We don't want to send it twice.
               !Cypress.isBrowser('firefox') ||
               // After Firefox 98,
-              // it sends a click event automatically if the element is a <button>
-              // it does not if the element is an <input>
-              (!isFirefoxBefore98 && $elements.isInput(event.target))
+              // it sends a click event automatically if the element is a <button>,
+              // but it does not if the element is an <input>.
+              // event.target is null when used with shadow DOM.
```

Comment:
How did this crop up during the Electron upgrade? Does this close an existing issue?

Reply:
The Firefox version in the bundled image was bumped to v98 from v93. There have been PRs over the last few months to fix some piece-mail issues with FF98, starting with https://github.com/cypress-io/cypress/pull/21042. This seems to be the last hold-out. We have similar checks in place in other type logic: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/actions/type.ts#L373